File: Exercism.io/python/scrabble-score/scrabble.py

scrabble_values = {'aeioulnrst': 1, 'dg': 2, 'bcmp': 3, 'fhvwy': 4,
                   'k': 5, 'jx': 8, 'qz': 10}

def score(word):
    return sum(map(get_scrabble_value, word.lower()))

def get_scrabble_value(char):
    for k in scrabble_values.keys():
        if char in k:
            return scrabble_values[k]
    return 0

File: Exercism.io/elixir/scrabble-score/scrabble.exs
defmodule Scrabble do
  @doc """
  Calculate the scrabble score for the word.
  """
  @spec score(String.t) :: non_neg_integer
  def score(word) do
    word |> String.downcase 
         |> String.codepoints
         |> Enum.map(&(get_letter_score(&1)))
         |> Enum.sum
  end

  def get_letter_score(letter) do
    cond do
      letter in ["a","e","i","o","u","l","n","r","s","t"] -> 1
      letter in ["d","g"]                                 -> 2
      letter in ["b","c","m","p"]                         -> 3
      letter in ["f","h","v","w","y"]                     -> 4
      letter in ["k"]                                     -> 5
      letter in ["j","x"]                                 -> 8
      letter in ["q","z"]                                 -> 10
      true                                                -> 0
    end
  end

end
